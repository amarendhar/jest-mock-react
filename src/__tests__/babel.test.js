import plugin from "../babel";
import { transform } from "@babel/core";

it("Creates mock single react component", () => {
  const source = `
  import React from 'react';
  import { render } from '@testing-library/react';
  import App from './App';
  import InnerComp from './InnerComp';
  
  mockComponents(InnerComp)
  
  test('Mock inner component', () => {
    const { getByText, debug } = render(App);
    debug();
  });
  
  `;

  expect(transform(source, { presets: ["react-app"], plugins: [plugin], filename: "/test.jsx" }).code).toMatchSnapshot();
});

it("Creates mock for multiple react component", () => {
    const source = `
    import React from 'react';
    import { render } from '@testing-library/react';
    import App from './App';
    import InnerComp from './InnerComp';
    import InnerComp2 from './InnerComp2';
    
    mockComponents(InnerComp, InnerComp2)
    
    test('Mock inner components', () => {
      const { getByText, debug } = render(App);
      debug();
    });
    
    `;

    expect(transform(source, { presets: ["react-app"], plugins: [plugin], filename: "/test.jsx" }).code).toMatchSnapshot();
});

it("Creates mock for react component named imports", () => {
    const source = `
    import React from 'react';
    import { render } from '@testing-library/react';
    import App from './App';
    import { InnerComp } from './InnerComp';
    
    mockComponents(InnerComp)
    
    test('Mock inner components', () => {
      const { getByText, debug } = render(App);
      debug();
    });
    
    `;

    expect(transform(source, { presets: ["react-app"], plugins: [plugin], filename: "/test.jsx" }).code).toMatchSnapshot();
});