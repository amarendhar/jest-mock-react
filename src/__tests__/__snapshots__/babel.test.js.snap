// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Creates mock for multiple react component 1`] = `
"\\"use strict\\";

jest.mock(\\"./InnerComp2\\", () => jest.fn(() => \\"InnerComp2\\").mockName(\\"InnerComp2\\"));
jest.mock(\\"./InnerComp\\", () => jest.fn(() => \\"InnerComp\\").mockName(\\"InnerComp\\"));

var _interopRequireDefault = require(\\"/Users/salahabdulsalam/Workspaces/jest-react-mock/node_modules/babel-preset-react-app/node_modules/@babel/runtime/helpers/interopRequireDefault\\");

var _react = _interopRequireDefault(require(\\"react\\"));

var _react2 = require(\\"@testing-library/react\\");

var _App = _interopRequireDefault(require(\\"./App\\"));

var _InnerComp = _interopRequireDefault(require(\\"./InnerComp\\"));

var _InnerComp2 = _interopRequireDefault(require(\\"./InnerComp2\\"));

test('Mock inner components', () => {
  const {
    getByText,
    debug
  } = (0, _react2.render)(_App.default);
  debug();
});"
`;

exports[`Creates mock for react component named imports 1`] = `
"\\"use strict\\";

jest.mock(\\"./InnerComp\\", () => ({
  InnerComp: jest.fn(() => \\"InnerComp\\").mockName(\\"InnerComp\\")
}));

var _interopRequireDefault = require(\\"/Users/salahabdulsalam/Workspaces/jest-react-mock/node_modules/babel-preset-react-app/node_modules/@babel/runtime/helpers/interopRequireDefault\\");

var _react = _interopRequireDefault(require(\\"react\\"));

var _react2 = require(\\"@testing-library/react\\");

var _App = _interopRequireDefault(require(\\"./App\\"));

var _InnerComp = require(\\"./InnerComp\\");

test('Mock inner components', () => {
  const {
    getByText,
    debug
  } = (0, _react2.render)(_App.default);
  debug();
});"
`;

exports[`Creates mock single react component 1`] = `
"\\"use strict\\";

jest.mock(\\"./InnerComp\\", () => jest.fn(() => \\"InnerComp\\").mockName(\\"InnerComp\\"));

var _interopRequireDefault = require(\\"/Users/salahabdulsalam/Workspaces/jest-react-mock/node_modules/babel-preset-react-app/node_modules/@babel/runtime/helpers/interopRequireDefault\\");

var _react = _interopRequireDefault(require(\\"react\\"));

var _react2 = require(\\"@testing-library/react\\");

var _App = _interopRequireDefault(require(\\"./App\\"));

var _InnerComp = _interopRequireDefault(require(\\"./InnerComp\\"));

test('Mock inner component', () => {
  const {
    getByText,
    debug
  } = (0, _react2.render)(_App.default);
  debug();
});"
`;
